core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1333,13 +1333,14 @@ public void populateDynamicChoices(ItemsetBinding itemset, TreeReference curQRef
                 new EvaluationContext(exprEvalContext, itemset.contextRef.contextualize(curQRef)));
 
         if (matches == null) {
-            throw new XPathException("Could not find references depended on by " + itemset.nodesetRef.getInstanceName());
+            throw new XPathException("No items found at '" + itemset.nodesetRef +
```

Comment:
If the issue that we expect to cause this is users not having access to the items in the table, I'm not sure that this error message quite captures that. Are you intentionally not mentioning that cause specifically?